core/src/Telemetry/TelemetryTokenCacheAccessor.cs

```diff
@@ -36,114 +36,74 @@ internal class TelemetryTokenCacheAccessor : TokenCacheAccessor
         // otherwise we would have to modify multiple implementations of TokenCacheAccessor on different platforms.
         public void SaveAccessToken(MsalAccessTokenCacheItem item, RequestContext requestContext)
         {
-            var cacheEvent = new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.AT };
-            CoreTelemetryService.GetInstance().StartEvent(requestContext.TelemetryRequestId, cacheEvent);
-            try
+            using (CoreTelemetryService.CreateTelemetryHelper(requestContext.TelemetryRequestId,
+                new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.AT }))
             {
                 SaveAccessToken(item);
             }
-            finally
-            {
-                CoreTelemetryService.GetInstance().StopEvent(requestContext.TelemetryRequestId, cacheEvent);
-            }
         }
 
         public void SaveRefreshToken(MsalRefreshTokenCacheItem item, RequestContext requestContext)
         {
-            var cacheEvent = new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.RT };
-            CoreTelemetryService.GetInstance().StartEvent(requestContext.TelemetryRequestId, cacheEvent);
-            try
+            using (CoreTelemetryService.CreateTelemetryHelper(requestContext.TelemetryRequestId,
+                new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.RT }))
             {
                 SaveRefreshToken(item);
             }
-            finally
-            {
-                CoreTelemetryService.GetInstance().StopEvent(requestContext.TelemetryRequestId, cacheEvent);
-            }
         }
 
         public void SaveIdToken(MsalIdTokenCacheItem item, RequestContext requestContext)
         {
-            var cacheEvent = new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.ID };
-            CoreTelemetryService.GetInstance().StartEvent(requestContext.TelemetryRequestId, cacheEvent);
-            try
+            using (CoreTelemetryService.CreateTelemetryHelper(requestContext.TelemetryRequestId,
+                new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.ID }))

```

Comment:
>new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.ID }) [](start = 16, length = 84)

suggestion: pass in only the params here and create the cache object in the helper method?
Will simplify the line. 
Perhaps also create a factory method for creating the cacheevent which can be used in the CreateTelemetryHelper method?

Reply:
The CacheEvent is not related directly to CreateTelemetryHelper, that class only knows about BaseEvent so I'm not sure it's appropriate to add knowledge of the specific events to the CreateTelemetryHelper method.  Maybe I'm missing your point here.

Reply:
Not good at explaining myself...
True, but given you are introducing a helper. Then I think we can help even more. Could be in another help where this is wrapped in though if that is better. I suppose a helper that took the full responsibility of creating the event and sending the event, and disposing it... is what I was looking for. No strong opinions on this though.

Reply:
Only the calling code knows which start/stop event(s) to actually log.  So I think it should be the calling code's responsibility to create the event and pass it in.

Reply:
But there could be a helper, now you are newing up another object to get this. my objectives are: 1. code simplicity and 2. SOC :) - No strong opinions btw. as this was a suggestion :)